packages/payment-proxy-client/src/App.tsx

```diff
@@ -149,15 +141,18 @@ export default function App() {
     // await nitroClient.WaitForObjective(result.Id);
 
     setPaymentChannelId(result.ChannelId);
-    updateChannelInfo(result.ChannelId);
-    console.timeEnd("Create Payment Channel");
 
-    // TODO: Slightly hacky but we wait a beat before querying so we see the updated balance
-    setTimeout(() => {
-      updateChannelInfo(result.ChannelId);
-    }, 1000);
-  };
+    nitroClient.PaymentChannelUpdated(result.ChannelId, setPaymentChannelInfo);
+
+    // It's possible the channel updated before we registered the handler above, so
+    // query the channel once now to get the latest information:
 
+    setPaymentChannelInfo(
+      await nitroClient?.GetPaymentChannel(result.ChannelId)
+    );
```

Comment:
This is a pretty common issue and makes for pretty bad DX. Perhaps a smarter event queue could be used in place of an event emitter.